crates/solver/src/solver.rs

```diff
@@ -483,6 +484,7 @@ pub async fn create(
             let score_calculator = score_configuration.get_calculator(solver_type);
 
             let solver = match solver_type {
+                SolverType::None => None,
```

Comment:
```suggestion
                SolverType::None => return None,
```
This enables a few improvements:
* makes pattern matching block further down obsolete
* don't have to return `Option` from `shared`
* no `.expect()` on `shared()` further down

Reply:
Ah of course, thank you!